cmd/argocd-application-controller/commands/argocd_application_controller.go

```diff
@@ -138,6 +139,9 @@ func NewCommand() *cobra.Command {
 				}
 			}))
 			kubectl := kubeutil.NewKubectl()
+			if shardingAlgorithm == "hash" {
```

Comment:
What if `shardingAlgorithm != "hash"` shouldn't we check that it is just a non empty string or nil? and set this value in Env ?

I think that we need to check only allowed values, and if they are valid, we should override the env var.
Otherwise, we let the env var do the job. 
wdyt ?